src/entities/tradeMining.ts

```diff
@@ -0,0 +1,155 @@
+  tradedVolume: Valuation;
+};
+
+type DefaultTradeMiningQUery = {
+  house: string;
+  phase: string;
+};
+
+type TradeMiningQuery = {
+  house: string;
+  phase: string;
+  numberOfTraders: number;
+};
+
+type GetUserRankQuery = {
+  house: string;
+  phase: string;
+  walletAddress: string;
+};
+
+type GetUserRankResponse = {
+  rank: number;
+  tradedVolume: Valuation;
+};
+
+export class TradeMining {
+  public methods(
+    signer: providers.JsonRpcSigner,
+    chainId?: number
+  ): Record<string, any> {
+    const networkInfo: NetworkInfo = distributeConstantsByNetwork(chainId);
+    const pendleSubgraphSdk = new PendleTradeMiningQuerySet(chainId || 43114);
+
+    const getTopTraders = async ({
+      phase,
+      house,
+      numberOfTraders = 10,
+    }: TradeMiningQuery): Promise<TradeMiningUser[]> => {
+      const topTraders = await pendleSubgraphSdk.getTopTraders(
+        phase,
+        house,
+        numberOfTraders
+      );
+
+      return topTraders.tradeMiningUsers.map((res: any) => {
+        return {
+          wallet: res.userAddress,
+          tradedVolume: {
+            currency: 'USD',
+            amount: parseFloat(res.volumeUSD).toFixed(2),
+          },
+        };
+      });
+    };
+
+    const getUserRank = async (
+      query: GetUserRankQuery
+    ): Promise<GetUserRankResponse> => {
+      const topTraders = await getTopTraders({
+        house: query.house,
+        phase: query.phase,
+        numberOfTraders: 1000,
+      });
+      let user = await pendleSubgraphSdk.getUserRank(
+        query.phase,
+        query.house,
+        query.walletAddress.toLowerCase()
+      );
+
+      if (user.tradeMiningUsers.length === 0)
+        return {
+          rank: -1,
+          tradedVolume: {
+            currency: 'USD',
+            amount: '0',
+          },
+        };
+
+      user = user.tradeMiningUsers[0];
+
+      let rank = 0;
+      for (const trader of topTraders) {
+        rank++;
+
+        if (trader.wallet === user.userAddress) {
+          return {
+            rank,
+            tradedVolume: trader.tradedVolume,
+          };
+        }
+      }
+      return {
+        rank: -1,
+        tradedVolume: {
+          currency: 'USD',
+          amount: parseFloat(user.volumeUSD).toFixed(2),
+        },
+      };
+    };
+
+    const getTotalTradedVolume = async (
+      query: DefaultTradeMiningQUery
+    ): Promise<Valuation> => {
+      const res = await pendleSubgraphSdk.getTotalTradedVolume(
+        query.phase,
+        query.house
+      );
+
+      if (res.tradeMiningHouses.length === 0) {
+        return { currency: 'USD', amount: '0' };
+      }
+
+      const tradeMiningHouse = res.tradeMiningHouses[0];
+
+      return {
+        currency: 'USD',
+        amount: parseFloat(tradeMiningHouse.volumeUSD).toFixed(2),
+      };
+    };
+
+    const getPhaseRewards = (query: DefaultTradeMiningQUery): TokenAmount => {
+      const pendleToken = new Token(
```

Comment:
Con't from my comment on line 45, if chainId is undefined, the PENDLE address will be the mainnet one. Since only Avax has trade mining, we may do a sanitary check on the chainId at the very beginning

Reply:
I think will leave this as is, ideally in the future this part of the SDK should support other chains from the get-go without any modification required i.e. sanitary check for Avax only